### PR TITLE
[FORMS-9369] implement slider and rating component

### PR DESCRIPTION
**Pull request by Aabha1**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/material-ui-based-components/src/components/Rating.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    const blurHandler = useCallback((event: any) => {
+        props.dispatchBlur(event.target.value);
+    }, [props.dispatchBlur]);
+
+    const focusHandler = useCallback((event: any) => {
+        props.dispatchFocus(event.target.value);
+    }, [props.dispatchFocus]);
+
+    return (
+        <Box>
```
Comment (amit-mnnit) [inline packages/material-ui-based-components/src/components/Rating.tsx:22]: can we use FormControl ?

File: packages/material-ui-based-components/src/components/Rating.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            <InputLabel error={isError} htmlFor={id} required={required}>
+                {label?.visible ? label.value : ''}
```
Comment (amit-mnnit) [inline packages/material-ui-based-components/src/components/Rating.tsx:24]: add this condition on InputLabel

File: packages/material-ui-based-components/src/components/Rating.tsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+            </InputLabel>
+            <Rating
+                id={id}
+                name={name}
+                value={value ? value : null}
+                disabled={!enabled}
+                readOnly={readOnly}
+                max={maximum ? maximum : 5}
+                onChange={changeHandler}
+                onBlur={blurHandler}
```
Comment (amit-mnnit) [inline packages/material-ui-based-components/src/components/Rating.tsx:34]: Does rating component provides the onBlur method ?

File: packages/material-ui-based-components/src/components/Slider.tsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+import { withRuleEngine } from '../shared/withRuleEngine';
+import { PROPS } from '../utils/types';
+import { Box, FormHelperText, InputLabel, Slider } from '@mui/material';
+
+const SliderComponent = (props: PROPS) => {
+    const { isError, name, label, required, description, enabled,
+        maximum, minimum, errorMessage, id, value, step }
+        = props;
+
+    function valuetext(value: number, label: any) {
```
Comment (amit-mnnit) [inline packages/material-ui-based-components/src/components/Slider.tsx:11]: bind with usecallback hook

Comment (amit-mnnit) [reply packages/material-ui-based-components/src/components/Slider.tsx:11]: rename the function name

File: packages/material-ui-based-components/src/components/Slider.tsx [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    const blurHandler = useCallback((event: any) => {
+        props.dispatchBlur(event.target.value);
+    }, [props.dispatchBlur]);
+
+    const focusHandler = useCallback((event: any) => {
+        props.dispatchFocus(event.target.value);
+    }, [props.dispatchFocus]);
+
+    return (
+        <Box>
```
Comment (amit-mnnit) [inline packages/material-ui-based-components/src/components/Slider.tsx:28]: can we use formcontrol?

**Review by amit-mnnit — CHANGES_REQUESTED** [review]
Please incorporate all the comments and also test localization

File: packages/material-ui-based-components/src/components/Rating.tsx (hunk 1 above)
Comment (Aabha1) [reply packages/material-ui-based-components/src/components/Rating.tsx:22]: No, on using FormControl, the layout is distorted and label overlaps over the rating component.

File: packages/material-ui-based-components/src/components/Rating.tsx (hunk 2 above)
Comment (Aabha1) [reply packages/material-ui-based-components/src/components/Rating.tsx:24]: Added on InputLabel

File: packages/material-ui-based-components/src/components/Rating.tsx (hunk 3 above)
Comment (Aabha1) [reply packages/material-ui-based-components/src/components/Rating.tsx:34]: Yes, it provides.

File: packages/material-ui-based-components/src/components/Slider.tsx (hunk 4 above)
Comment (Aabha1) [reply packages/material-ui-based-components/src/components/Slider.tsx:11]: Binded with useCallback hook and changed the name to getValue.

File: packages/material-ui-based-components/src/components/Slider.tsx (hunk 5 above)
Comment (Aabha1) [reply packages/material-ui-based-components/src/components/Slider.tsx:28]: No, on using FormControl, the layout is distorted and label overlaps over the slider component.

**Pull request merged**